unified-installer/astra-unified-installer.sh

```diff
@@ -2133,136 +2340,14 @@ step_init_generated_dirs_and_files() {
-    local -r connector_namespace="$(get_connector_namespace)"
-    local -r trident_namespace="$(get_trident_namespace)"
-    local -r connector_registry="$(join_rpath "$CONNECTOR_IMAGE_REGISTRY" "$(get_base_repo "$CONNECTOR_IMAGE_REPO")")"
-    local -r trident_acp_registry="$(join_rpath "$TRIDENT_ACP_IMAGE_REGISTRY" "$(get_base_repo "$TRIDENT_ACP_IMAGE_REPO")")"
-    local -r encoded_creds=$(echo -n "$ASTRA_ACCOUNT_ID:$ASTRA_API_TOKEN" | base64)
-
-    [ -z "$kustomization_file" ] && fatal "no kustomization file given"
-    [ ! -f "$kustomization_file" ] && fatal "kustomization file '$kustomization_file' does not exist"
-
-       # SECRET GENERATOR
-    cat <<EOF >> "$kustomization_file"
-generatorOptions:
-  disableNameSuffixHash: true
-secretGenerator:
-- name: astra-api-token
-  namespace: "${connector_namespace}"
-  literals:
-  - apiToken="${ASTRA_API_TOKEN}"
-EOF
-    if [ -z "$global_pull_secret" ]; then
-        # if image pull secret is empty, set same name for connector and trident secret so torc patch works as expected
-        IMAGE_PULL_SECRET="astra-regcred"
-        if components_include_connector; then
-            cat <<EOF >> "$kustomization_file"
-- name: "${IMAGE_PULL_SECRET}"
-  namespace: "${connector_namespace}"
-  type: kubernetes.io/dockerconfigjson
-  literals:
-  - |
-    .dockerconfigjson={
-      "auths": {
-        "${connector_registry}": {
-          "username": "$ASTRA_ACCOUNT_ID",
-          "password": "$ASTRA_API_TOKEN",
-          "auth": "${encoded_creds}"
-        }
-      }
-    }
-EOF
-            logdebug "$kustomization_file: added connector secret to namespace $connector_namespace"
-        fi
-    
-        if components_include_trident && [ "$trident_namespace" != "$connector_namespace" ]; then
-            cat <<EOF >> "$kustomization_file"
-- name: "${IMAGE_PULL_SECRET}"
-  namespace: "${trident_namespace}"
-  type: kubernetes.io/dockerconfigjson
-  literals:
-  - |
-    .dockerconfigjson={
-      "auths": {
-        "${trident_acp_registry}": {
-          "username": "$ASTRA_ACCOUNT_ID",
-          "password": "$ASTRA_API_TOKEN",
-          "auth": "${encoded_creds}"
-        }
-      }
-    }
-EOF
-            logdebug "$kustomization_file: added trident acp secret to namespace $trident_namespace"
-        fi
-    fi
-
-    insert_into_file_after_pattern "$kustomization_file" "patches:" '
-- target:
-    kind: Deployment
-  patch: |-
-    - op: replace
-      path: /spec/template/spec/imagePullSecrets
-      value:
-        - name: "'"${global_pull_secret}"'"
-'
-    logdebug "$kustomization_file: added pull secret patch ($global_pull_secret)"
+    generate_kustomization_dirs_and_files "$kustomization_file"
 }
 
 step_generate_astra_connector_yaml() {
-    local -r kustomization_file="$__GENERATED_KUSTOMIZATION_FILE"
-    local -r kustomization_dir="$(dirname "$kustomization_file")"
+    local -r connector_kustomization_file="$__GENERATED_CONNECTOR_KUSTOMIZATION_FILE"
```

Comment:
This step no longer operate on the main kustomization file, only the connector kustomization file